### ISSUE TITLE
How to make multipe components (web-services) share an ingress instance?

### ISSUE THREAD

**Issue by ghostbody** [kubevela/kubevela#1378]
Suppose I have multiple web-service components, and many of them need a route trait to expose the ports.

```yaml
name: my-web-app

services:
  web-service1:
    type: webservice
    image: oamdev/testapp1:v1
    cmd: ["node", "server.js"]
    port: 8080
    cpu: "0.1"
    route:
      domain: example.com
      issuer: tls
      rules:
        - path: /web-service1
          rewriteTarget: /

  web-service2:
    type: webservice
    image: oamdev/testapp2:v1
    cmd: ["node", "server.js"]
    port: 8080
    cpu: "0.1"
    route:
      domain: example.com
      issuer: tls
      rules:
        - path: /web-service2
          rewriteTarget: /

```

It seems like kubevela will create as many ingress instance as services. How can I acheive one ingress instance which is shared by all these components?

**Comment by xiqiu618** [comment]
I can try to improve this ability, please assign to me
/assign

**Comment by wonderflow** [comment]
@xiqiu618 can you explain something about your solution

**Comment by xiqiu618** [comment]
after i thinking about it,  separate create ingress instance  for each component. can be better managed ,merge rules by host may be more affect.

**Comment by wonderflow** [comment]
> after i thinking about it, separate create ingress instance for each component. can be better managed ,merge rules by host may be more affect and complex

Yes, that's why we don't support this case. And I don't even know why we need to use one ingress instance to share? It doesn't cost anything to create multiple instances.

**Comment by ghostbody** [comment]
In my case, some services need the same entrypoint (like traditional reverse proxy server).

i.e. some-product.kubevela.com -> same ingress instance -> multiple services

**Comment by wonderflow** [comment]
Does these services belong to the same application?

**Comment by wonderflow** [comment]
Check the policy for this feature： https://kubevela.net/docs/platform-engineers/policy/custom-policy

It's implemented in v1.4
